### ISSUE TITLE
Keep transitive deps that would look optional after pruning

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [jplatte/cargo-depgraph#15]
Locally I wrote some code to generate the image below. This deduplicates transitive dependencies, but keeps those that 'go around' optional deps. I.e., the solid line between `pa-web` and `pa-vis` is nice because without it (the current behavior) it would seems as though a large part of the tree is optional, while in fact it's not.

Would you accept a PR for it? And in that case, what should the flag be named? I'm not sure myself. And should this be default?

![depgraph](https://user-images.githubusercontent.com/6233682/226058287-f4ef8330-0771-4964-bd8c-08e468f62640.svg)

**Comment by jplatte** [comment]
I think it makes sense for this to be the default behavior of `--dedup-transitive-deps`, not just for optional dependencies, but target-specific dependencies as well.

**Comment by RagnarGrootKoerkamp** [comment]
Yes I agree. Then the question becomes how to handle optional and target-specific dependencies when there is a path of optional/target-specific dependencies.

I think optional deps should still be shown, but target-specific deps can be inferred through each other.

One issue with this could be when a long chain of packages has a similar feature flag to enable one specific dependency (say `clap`), and then we draw multiple optional lines to that package. On the other hand, drawing the dependency only for the bottom one isn't complete either.
